src/transformers/trainer.py

```diff
@@ -1297,17 +1350,32 @@ def prediction_loop(
 
         return PredictionOutput(predictions=preds, label_ids=label_ids, metrics=metrics)
 
-    def distributed_concat(self, tensor: torch.Tensor, num_total_examples: int) -> torch.Tensor:
+    def distributed_concat(self, tensor: torch.Tensor, num_total_examples: Optional[int] = None) -> torch.Tensor:
         assert self.args.local_rank != -1
 
         output_tensors = [tensor.clone() for _ in range(torch.distributed.get_world_size())]
         torch.distributed.all_gather(output_tensors, tensor)
+        concat = torch.cat(output_tensors, dim=0)
+
+        # truncate the dummy elements added by SequentialDistributedSampler
+        if num_total_examples is not None:
+            concat = concat[:num_total_examples]
+        return concat
+
+    def distributed_broadcast_scalars(
```

Comment:
This doesn't seem to use self (distributed_concat neither) so maybe those two methods should be functions?

Reply:
Agree and will move them out, do you think we should keep a redirection to keep `distributed_concat` backwards-compatible?